client.go

```diff
@@ -530,7 +530,7 @@ type DNSData struct {
 	MX             []string   `json:"mx,omitempty"`
 	PTR            []string   `json:"ptr,omitempty"`
 	ANY            []string   `json:"any,omitempty"`
-	SOA            []string   `json:"soa,omitempty"`
+	SOA            SOA        `json:"soa,omitempty"`
```

Comment:
According to [rfc1034](https://datatracker.ietf.org/doc/html/rfc1034), there should be only one SOA answer record, but the library tends to tolerate non-compliant servers. So I'd suggest changing it to:

```go
type DNSData struct {
...
  SOA []SOA
...
}
```